src/android_service.py

```diff
@@ -37,5 +41,25 @@
 thread.daemon = True
 thread.start()
 
-while True:
-    time.sleep(1)
\ No newline at end of file
+# start a parallel Flask server as a backchannel for triggering events
+flaskapp = flask.Flask(__name__)
+
+@flaskapp.route('/share_by_intent')
+def do_share_by_intent():
+
+    args = flask.request.args
+    allowed_args = ["filename", "path", "msg", "app", "mimetype"]
+    kwargs = {key: args[key] for key in args if key in allowed_args}
+
+    if "filename" in kwargs:
+        kwargs["path"] = get_content_file_path(kwargs.pop("filename"))
+
+    logging.error("Sharing: {}".format(kwargs))
+
+    share_by_intent(**kwargs)
+
+    return "<html><body style='background: white;'>OK, boomer</body></html>"
```

Comment:
Ship it! :)

Reply:
@jamalex More seriously, why do we return an HTML page rather than a JSON status result? I'm guessing it's ignored by Kolibri, but we probably don't want some weird bug in Kolibri to cause this page to get shown... ;-)